### DISCUSSION TITLE
balance patch and more realistic game logic

### DISCUSSION THREAD

**Discussion by Sadret** [OpenRCT2/OpenRCT2#17623] [Feature requests]
### Scope

This discussion aims to collect ideas to balance or improve the game logic in places where the current game logic is nonsensical, vastly imbalanced, or easily exploitable. The end goal is to implement the new logic in parallel to the old logic, and let the _player_ choose which logic to use when they _start a new game_ (it might also be possible to switch between the two implementations during a running scenario).

**Note**: This discussion is _not_ about introducing _new_ game features and mechanics, but about improving exiting ones.

When fixing the issues discussed here, we should follow these rules:
- Compared to the old logic, the new logic should be closer to what an inexperienced player would naturally expect.
- The new logic does not introduce new or change existing UI elements. This also excludes for example new peep or ride stats.
Thus, only experienced players that are already familiar with the game would notice these changes.

**Contribute**: If you want to make a suggestion, please feel free to participate in the discussion. I will update the following list accordingly. If an issue is already addressed, but you want to improve the suggested solution, feel free to do so. If you think that an issue is not important or that the majority of players prefer the current logic, and thus the logic should be unchanged, then you can comment on that, but I will keep the issue as long as it still fits the scope of this discussion.

### Guest Behaviour
- There is an optimal ride price which is simply the maximum that a eligible guest will pay.
  - In a certain price range, _all_ eligible guests will go on a ride, above that threshold, _no_ guest will go on that ride.
  - Reducing the ride price should increase demand.
  - Solution: Implement a demand curve. The lower the price, the more guests are willing to go on a ride.
- Queue length has no influence on the willingness of a guest to join a queue.
  - Solution: The longer the queue, the less likely it is that a guest will join a queue.
  - Note: This may be seen as a new mechanic instead of improving an old one, _but_: Guests are already complaining about long queue lines and this is very visible to the player (e.g. park messages and guest thoughts, also queue length and queue time are already stats in the 'rides' window). Thus, the naive thing to assume is that guests will actively avoid long queue lines.
- Queue line TVs and entertainers will preserve a guest's happiness forever.
  - On the other hand, it would make sense if nearby scenery and tracks would affect queueing guests, but it does not. Also, guests that are currently eating or drinking should get bored slower.
- Guests will pay anything (which means 20€ without cheats) for an umbrella when it's raining.
  - While that price seems fair compared to real world umbrella prices, this is very imbalanced when considered in the setting of this game.
  - Solution: Guests will pay slightly more for umbrellas while it is raining, but not any price. Guest happiness could decrease a bit if umbrellas are too expensive when it rains.

### Rides
- Micro coasters are too powerful. They have unreasonably high stats and attract too many guests.
  - While some micro coaster design are sensible, most of them are not. In the end, a micro coaster should perform similar to a comparable (e.g. in speed and ride length) flat ride (e.g. free fall). For example, the standard Station+Station+HalfCorkScrew launched coaster should perform similar to a launched free fall with the same launch speed (i.e. poorly) (I was made aware that this comparison is bad because micro free falls are also too powerful, see below).
  - Solution: Remove that rides with inversions are excepted from minimum requirement penalties.
See also the point about short-timed flat rides and the one about multiple ride types.
- The free fall and possibly some other short-timed flat rides are too powerful.
  - While it seems sensible that a free fall should have high excitement and intensity rating, that does not mean that it should be worth the same as a 60-second coaster with the same stats.
  - Solution: When calculating the ride value, include other stats, e.g. ride time. For example, a 2% penalty for every second that the ride is shorter than 30 seconds.
See also the point about multiple ride types.
- The ride bonus value only depends on the ride type and not on the ride design, e.g. ride stats or ride value.
- A high nausea rating increases the ride value. Also, excessive intensity stats still increase the ride value.
  - Solution: Implement a value curve for each stat that first rises and the drops off.
- While most excitement boosts make sense, some of them are just a box checking exercise, i.e. synchronized stations, adjacent scenery, adjacent paths / headchopper, interlocking loops.
  - Problem: It might be very difficult to find out when an element is placed only for the excitement boost.
- Multiple rides of the same type are penalised too little with respect to ride value and ride bonus value.
  - While slightly overpowered or just strong rides ar not a problem in itself, spamming them makes them really overpowered.
  - Solution: Implement diminishing returns, e.g. every consecutive ride achieves only 80% of the stats as the previous one of the same type.
- The price decay for aging coasters is too high.

### Park entrance fee, Ads and Guest Generation
- Ads are too strong in general.
  - In both pay-for-entry and pay-per-ride parks, the amount that the ad-generated guests spend in your park usually is much more than the cost for advertisement. Thus, in most situations there is no reason not to have ads, except for if you are totally broke, which should not happen (see also the point about loans).
  - On the other hand, a good ad campaign _should_ make a positive profit, so it is debatable if this is really an issue or how to properly fix it.
- Ads are much too strong in small parks with few or even no rides.
  - Ads increase the guest generation but ignore the park entrance fee.
  - Solution: punish the guest generation by ads if the park entrance fee is too high.
- 2-week ads let you advertise for nearly 3 weeks while paying for only 2.
  - Solution: Stop ads after exactly two weeks.
See also the points about park entrance fee.
- The maximum park entrance fee that you can charge increases too quickly with the build rides.
  - A very small park can already charge the maximum fee.
  - A guest pays for the attractions that are present when they enter the park, but then can enjoy everything that is build and opened afterwards. A lower entrance fee withholds the player from income that he may well deserve.
- Harder guest generation actually makes it much easier to attract guests if you are able to build good coasters, while at the same time everything else contributes next to nothing.
- In the real world, parks close at the end of the day. Implement a system that makes the guests leave after some time.
  - In pay-for-ride parks, guests will leave after they spent most of their money. Regarding balancing, this needs little to no change.
  - In pay-for-entry parks, having a good park punishes the player, as guests are much less likely to leave it, which hurts the income a lot. Thus, these parks would benefit a lot from the end-of-day mechanism.
  - Possible solutions (should be implemented as soft constraints):
    - Make guests leave after a certain amount of time. It is reasonable that guests stay longer in a bigger park, so the time might need to scale with park size.
    - Make guests leave after they enjoyed a certain amount of rides. Similar to the time constraint.
    - Make guests leave after they enjoyed rides of a certain accumulated value, e.g. the entrance fee that the guest payed.
      - Problem: makes pay-for-entry and pay-for-ride parks virtually the same.
  - Problem: It will be much more difficult to increase the amount of guests in the park.
    - Solution: Adjust guest generation to account for more guests leaving the park. This probably requires a lot of balancing and fine-tuning to not make pay-for-entry parks absolute money-makers.

### Miscellaneous
- In money-free scenarios, you can have a (nearly) unlimited amount of staff, which completely removes certain aspects of the game, e.g. keeping your paths clean or your guests happy.
  - Solution: Tie the amount of staff that can be hired to e.g. the current amount of guests or the soft guest cap. E.g. 1 staff member / 50 guests + 5 staff members.
- Loan interests are more or less irrelevant up to the point were taking up a loan is a no-brainer in allmost all situations.
  - Getting a loan in a 4-year scenario at a rather high interest rate of 10% (most scenarios have much less) means that 60% of the loan is "for free".
- Cash machines are too strong in pay-for-ride parks, as they let you make tons of money while keeping all guests in your park.
  - Solution: Limit to one use per guest, which is still very strong.
  - Solution: Limit to certain amount of money per use, and make it easier to please guests with each withdrawal.

### Out of Scope
- Peep pathfinding and path layouts: This is a huge topic that is already worked on and thus should be handled as a separate issue.

**Comment by James103** [comment]
> A high nausea rating increases the ride value.

Possible solution: Implement a curve for the effect on a ride's stats on its value. For example, every additional point in Nausea after the first 10 reduces ride value by some amount (and a similar penalty for excessive Intensity).

**Reply by Sadret** [reply]
Noted

**Comment by MarcelVos96** [comment]
> For example, the standard Station+Station+HalfCorkScrew launched coaster should perform similar to a launched free fall with the same launch speed (i.e. poorly).

I have some bad news for you as a tiny launched freefall is arguably even more overpowered than the microcorkscrew.

That aside, I think that this is a good idea, but only if it is NEVER the default version of the game logic. This should be a checkbox somewhere in the options that is something like "Enable alternate game logic". Most OpenRCT2 users have already played either vanilla or classic, and the game suddenly behaving very differently from what they know is going to alienate them.

Currently there are a lot of guides out there on how the game logic works, of which quite a few were made by Deurklink and me. It would be a very bad move to make all those guides suddenly invalid by changing the default behaviour of the game. Players would look up those guides and other info and follow them, but it won't work as they're not accurate anymore.

**Reply by Sadret** [reply]
As said, "let the player choose which logic to use when they start a new game". Either by an option in the scenario select window (which I would prefer), or by a global option in the options window (which will be forgotten or never discovered).

**Reply by MarcelVos96** [reply]
It was still a bit ambiguous so I wanted to make sure as this isn't something that should be done lightly. A checkbox somewhere on the scenario selection screen would be fine too, as long as it doesn't become a popup that you get whenever you open up a new scenario.

**Reply by Sadret** [reply]
I agree, that's too annoying. Maybe similar to this checkbox here:
![image](https://user-images.githubusercontent.com/15895532/181282398-bcdbc512-caee-46d9-9491-271140f05db6.png)

**Reply by ocalhoun6** [reply]
Perhaps _some_ of these suggestions should become default aspects of the game, as just a bit of much needed rebalancing. (Such as the demand curve for ride prices, instead of a hard cutoff.)

While other aspects of what's being discussed here might change things a lot more and should be optional. (Such as the restriction on how much staff you can hire in a no-money park.)

**Reply by Sadret** [reply]
I mostly disagree, I think that already changing the price/demand behaviour is too much of a change, especially considering that it may break existing parks.

**Comment by duncanspumpkin** [comment]
I would call it a "Balance patch" rather than improved logic.

**Comment by James103** [comment]
> Multiple rides of the same type are penalised too little

See also https://github.com/OpenRCT2/OpenRCT2/discussions/17042

Main ideas from that discussion:
* For non-tracked rides,
	* Each instance after the first (of a given ride type) contributes less to the soft guest cap.
	* The more instances of that ride type, the lower the ride value for each instance.
* For tracked rides,
	* If the game detects two tracked ride designs that share some of the same characteristics, both rides are penalized.
	* A couple of stat requirements should be added to further penalize microcoasters.

**Reply by Sadret** [reply]
Most of the ideas are included now.
As already discussed elsewhere, checking for "similar" track designs is very difficult and I would put it out of scope.

**Comment by James103** [comment]
> Guests will pay anything (which means 20€ without cheats) for an umbrella when it's raining.

Possible solution: When it is raining, if a guest would not ordinarily buy an umbrella, there is a price- and happiness-dependent chance the guest will change their mind and buy it anyway due to the weather (even more so when it's thundering). If the umbrellas sold there are too expensive, guests will never buy them regardless of happiness, weather, and their money in purse.

In other words, guests should purchase umbrellas more often and may be willing to pay slightly, but not significantly / game-breaking, more for an umbrella when it is raining.

**Comment by ocalhoun6** [comment]
>Queue line TVs and entertainers will preserve a guest's happiness forever.

Along these lines, I remember the original RCT1's instruction booklet mentioning things like, "Put some trees next to the queue line, guests will be happy to have some shade."

But the actual game has no such mechanic to it -- guests don't care about scenery at all unless it's influencing ride stats or unless there's a _lot_ of scenery and they get the 'great scenery' thought.

I think it would be intuitive (and ironically enough go with the original games instruction booklet) to give peeps a _very small_ happiness boost if there are scenery items very close to the queue line the peep is in. (Say, within a 3x3 square around them.)

And that makes sense from a real-world perspective. A long queue is more bearable if there's nice scenery around it, whether that's flowers and trees or complex set pieces for a themed ride. Many real-world parks have elaborate decorations around their queue lines to help entertain guests as they wait. -- [Example 1](https://vignette4.wikia.nocookie.net/disney/images/d/d3/Star_Tours_Queue_Line_facing_Starspeeder.jpg) -- [Example 2](https://i1.wp.com/www.disneytouristblog.com/wp-content/uploads/2017/08/radiator-springs-racers-queue-disney-california-adventure.jpg) -- [Example 3](https://i.ytimg.com/vi/KuJTXu6MaU8/maxresdefault.jpg)

There might be a similar bonus for having ride track within the same distance. Other real world rides use interactions between the ride track and the queue line to make waiting in the queue more interesting. (See example 3 above.)

-----

Might also consider having guests' happiness decrease more slowly in queue lines if the guest is currently eating or drinking something, as that would occupy them somewhat and make the wait less onerous.

**Reply by MarcelVos96** [reply]
> There might be a similar bonus for having ride track within the same distance. Other real world rides use interactions between the ride track and the queue line to make waiting in the queue more interesting. (See example 3 above.)

A bonus for that does exist, but it's for the ride. The excitement rating of a ride is increased a bit if there is a queue line near its track (doesn't have to be its own).

**Reply by ocalhoun6** [reply]
Well, yes, there's that of course. But I mean also a small happiness bonus to the peeps waiting in the line, since it would entertain them somewhat to watch the ride go past them, and maybe build up anticipation of going on the ride yourself.

**Reply by Sadret** [reply]
I included the idea, but it might be too much to implement.

**Comment by Basssiiie** [comment]
While I understand the wish for changing the balance, I don't think the base game balance should be replaced:
- people will dislike it just because it got changed.
- people will never agree on the right kind of balance changes.
- guides will be invalidated.
- previous experiences will be invalidated.
- it may kill nostalgia or the original feeling of the game.
- it would heavily change scenario balance, both base game and custom scenarios.
- it would be very hard to keep the difficulty somewhat the same.

My suggestion would be to expand the plugin api to such extend, that it will be possible to create balance plugins that override or replace base game behaviour. 

### The advantage of balance plugins:
- players are not limited to one set of rebalance rules.
- players could choose between different kinds of balance plugins, for example;
   - hard mode 
   - easy mode
   - realistic balance (could make the guests behave more realistic, mimic real world behaviour)
   - some kind of gamified challenge balance (could make the game more strategic, where choices always have both advantages and disadvantages)
- players can pick the type of balance that interests them, or stick to the base game if they prefer that.
- players could switch to another balance plugin at any time.

### To make this happen, the plugin api could be expanded with things like:
- allow control over guest goals, for example:
    - is the guest heading for a ride?
    - is the guest heading for the park exit?
    - is the guest heading to a toilet?
- allow modifying the guest thoughts, to influence whether they think a ride queue is too long or things alike.
- allow properties or callbacks for guest choices:
   - should the guest join the queue?
   - does the guest like the ride?
   - should the guest buy the item?
- allow a way to influence the spawn rate, either though modifying the spawn percentage or a callback.

(Note: these are example ideas, not definitive preferred implementations)

### Some things that are already possible in the plugin api:
- changing a guests happiness, energy, hunger, thirst and toilet needs.
- changing a ride's excitement, intensity and nausea values.
- changing the ride's park value.

**Reply by Sadret** [reply]
First, I made it very clear that the changes should not replace the current logic, but co-exist with them side-by-side, keeping the old logic as the default. This makes your first paragraph irrelevant.

Second, I do not see why the advantages you list only apply for plugins and not for a native implementation. The only advantage I see is that it is easier for other people to contribute, as releasing a plugin is much easier than getting a PR through.

Anyway, feel free to implement the needed API functionality. I would be happy if this project could actually be realised as a plugin, but I think most of the suggested changes are difficult to do in that way.

**Reply by Basssiiie** [reply]
Ah I missed the one line that mentions the options when you start the scenario, my bad. Still, new players would have no clue what "original" and "improved" logic really entails. So some of the options, like invalidating guides would still stand. Furthermore, it would also invalidate plugins that go about detailing the current implementation, unless they're updated to keep track of every native implementation that gets added (this includes both your ride price manager and soft guest cap calculator). It may make both these guides and these plugins also prone to spam like "it doesn't work as expected" because players are not aware on which logic option they are. They would be likely be more aware if they have actively installed a specific logic balance plugin.

> The only advantage I see is that it is easier for other people to contribute, as releasing a plugin is much easier than getting a PR through.

Yes you're right, every logic variation would require a PR and would require review from the developers. I think it would also be quite a challenge to make sure all these custom logic variations do not cloud the source code with `if (logic == "realistic")`-like checks everywhere. Though if the implementation is required to be more abstract, then it would also be easier to just hook it into the plugin api I suppose.

Making plugins is also way more accessible for anyone than adding things to the source code:
- the programming language is easier to use;
- a basic development environment is easier to set up, especially on non-Visual Studio setups;
- the code base you need to get familiar with is way smaller (plugin api vs. the game's source code)

> Anyway, feel free to implement the needed API functionality. I would be happy if this project could actually be realised as a plugin, but I think most of the suggested changes are difficult to do in that way.

Both native and plugin functionality could be done in small steps (should be done in small steps), to implement features one by one. And I suppose anyone can feel free to try either implementation.

**Comment by frutiemax** [comment]
OpenTTD did something similar by allowing different game rule sets.

**Comment by X123M3-256** [comment]
> Solution: Remove that rides with inversions are excepted from minimum requirement penalties.

I would just ditch minimum stat requirement penalties altogether, they make no sense. Adding an extra 5ft in height shouldn't cause large changes in the excitement rating. I think that trying to neuter specific exploits by patching the existing game logic is the wrong way to go about it: the game should be made to function in a logical way so that the correct behaviour naturally emerges.

I would do away with the existing ride rating calculation altogether, and replace it with something much more sophisticated. Being able to accurately assess the value of rides is key to a game like this, and a lot of the problems stem from how poorly the game does this.

> Problem: It might be very difficult to find out when an element is placed only for the excitement boost.

I think that this problem is insurmountable if you want to keep the boost, but it's not necessarily a problem - it still costs money to place those dummy pieces, so the player isn't really gaining an advantage from doing it ... they could just as well do it properly. I don't think you can or really should force the player to build realistically, I just don't think they should be penalized for doing so.

The synchronized station bonus, if you wanted to get fancy, could look at the actual proximity of the two trains during the test run. I think if you're going to do this, the bonus should be increased, as building a ride that synchronizes well is fairly difficult and should be rewarded.

> Solution: Limit to one use per guest, which is still very strong.

I don't like this solution because it's not how real cash machines work ... instead, I'd propose a soft cap on the total amount a guest is willing to spend, after which they're more and more likely to leave the park.

> In money-free scenarios, you can have a (nearly) unlimited amount of staff, which completely removes certain aspects of the game, e.g. keeping your paths clean or your guests happy. 

Eh, in money free scenarios you can have unlimited amounts of everything. You can completely flatten the terrain in Extreme Heights or Lucky Lake and cover it with coasters. The solution here is that the money free option should only really be used for easy scenarios or those intended to be more of a sandbox ... if the scenario creator has decided to make everything unlimited, I don't really think we need to impose an arbitrary cap.

I'd say the major point you haven't mentioned is the ride rating function: it's awful, and while I certainly agree that we should make things like the bonus value dependent on the ride type, that won't necessarily have the desired effect if the ride rating function itself doesn't accurately assess the value of the ride. Microcoasters should be poor performing not because we add a penalty for small rides, but because they get a low excitement in the first place from being so short. As long as ride ratings remain a simple box checking exercise, they'll continue to be exploitable.

**Reply by MarcelVos96** [reply]
Considering how massive a task new ride stat calculations are they might be better as a seperate thing. As long as normal coasters still return similar stats as they do now this shouldn't mess with the balance of everything else too much and could stand on its own.

The stat calculations are quite complicated and the more realistic ones will probably be even more complicated, so it's likely going to be more work than every other balance change put together.

I think a plugin called "realistic ride stats" would work quite well actually.

**Reply by X123M3-256** [reply]
> The stat calculations are quite complicated 

They really aren't, they're absurdly simple. When testing the ride rating functions for the various ride types I've built I found that the layout is pretty much irrelevant - if you're not going out of your way to exploit the ride rating function, you always get stats that are very similar. The only thing that really seems to make a difference is the base stats for the ride type, which I don't think should be a thing in the first place, and the ride length as a distant second.

I don't want normal coasters to return stats similar to those that they do now. I don't think any part of the existing ride rating calculation is fixable, I think it's hopelessly broken and that it should be replaced altogether as the first step of any rebalancing effort.

**Reply by MarcelVos96** [reply]
What's wrong with for example the 8.3/7.0/3.4 of Goliath in Six Flags Magic Mountain? Those seem like decently realistic stats and in whatever new stat system you come up with it should give somewhat similar stats. If coasters like that suddenly only have 5 excitement or have 12 excitement, the new stat system is going to feel extremely random.

Oh and with more complicated I mean that a lot of factors are going into it compared to the other stuff. The algorithm for guest generation or the maximum entry fee is many times simpler than the EIN stat algorithms.

**Reply by Sadret** [reply]
> They really aren't, they're absurdly simple.

Are they? As far as I remember the ride rating function iterates over the track and counts inversions, turns, and other stuff.

**Reply by X123M3-256** [reply]
> What's wrong with for example the 8.3/7.0/3.4 of Goliath in Six Flags Magic Mountain? 

Nothing is wrong with those stats. If you take any ride type and build a good layout that's realistic for that ride type (maybe not the heartline twister), you'll get decent and fairly representative stats. The problem is that you'll still get pretty much the same stats if you build *any* layout, as long as it's roughly the same length and meets stat requirements.

See, for any given layout, it shouldn't get more excitement simply because it's an Intamin mega and not a Schwarzkopf looper. That ride type should get high excitement because of its height and speed, and it should *only* get high excitement if the layout actually takes advantage of that. Given identical layouts, there should really be no difference in the stats of the Intamin and the Schwarzkopf. But there will be, because the game gives each ride a base excitement value that seems to matter more than anything else. And this is the primary reason why microcoasters are so exploitable, because as long as you can hit all the stat requirements the actual layout matters little, and can therefore be very short.

When I implemented the ride rating function for the hybrid coaster, my thought process went like this. Overall, I want this ride type to get similar ratings to the giga coaster. But, it should prioritize different things - there should be more of an emphasis on turns and inversions. I wanted it to be such that, if you built a simple out and back layout with airtime hills like you might with the giga, you'd get good but not great stats - you're not taking full advantage of the ride types' capabilities.

So, I made it so that the turns, G forces, and inversions were weighted more heavily than the giga and the airtime and average speed was weighted less. I'd hoped that this would make it so that a well-designed hybrid gets ratings that are similar, but if you just take a giga layout and swap the ride type, the ratings would go down (and vice versa if you build a hybrid layout and then make it a giga).

What I found is that it the ratings would only vary by like 0.1. I actually found it very hard to design a layout that would get significantly different ratings as a giga than it would as a hybrid, even knowing exactly how the two rating functions were implemented and where they differed. I was able to get it so that a hypercoaster type layout would give you *slightly* less excitement on the hybrid than it would on the giga, but things like the maximum G force will be pretty similar for any well designed layout, and those were the areas where the rating functions actually differed. By far the biggest influence on the rating was the ride types base excitement, and then the total ride length.

> Are they? As far as I remember the ride rating function iterates over the track and counts inversions, turns, and other stuff.

It does, but when it comes to calculating the actual stats it just uses those aggregate statistics. It doesn't matter if you have one high G turn or 10, because only the max G force is considered. It does take account of average speed, but this includes the lift hill so changing the lift speed can effect the excitement rating way more than it should. I would iterate over track pieces and calculate a stat contribution per track piece. That way, the actual layout matters more, and it's more intuitive. I often get people asking how my new track pieces affect stats, because it seems logical that they would ... but they don't, except indirectly by increasing the inversion count. (I think the river rapids special pieces are an exception in that they directly add excitement, and there may be others but it's not the norm, and I think it should be).

**Reply by TheAtomTwister** [reply]
One thing that should be taken into consideration when adding a totally new ride stat calculation system is how the real world counterparts of the ride feel.  The old school Arrow track's design is more prone to a kind of roughness that tends to add a bit more intensity and can become great enough to subtract from the overall enjoyment factor of the ride.  There come into play some engineering nuances with aging arrows and fatigued metal that cause the intensity to increase (due to the increasing roughness) because no longer does the track gently hug the train keeping the ride smooth but now is too loose around the train which leads it to be battered around such as is what happened to Vortex.  Refurbishing the ride would fix this.

Some coasters have track designs that are inherently rougher than others, some are much more resistant to metal fatigue over time.  Intamin and B&M coasters for example have track design to where the train does not stress the track, causing minimal metal fatigue which in turn leads to the ride remaining smooth.  

So, when adding a new ride stat calculation system, these factors will need to be considered.  Staying well connected to folks like Theme Park Review and other people who are familiar with the real rides can really help when working to find out what the appropriate algorithms are.

**Reply by karst** [reply]
One of the issues is for example the arrow track is used by vekoma and a few others aswell. Old vekomas were quite similar to arrows in tracking. But a new chance ride (like you find in Kentucky Kingdom) is butter smooth. Not to mention redos of SLCs which actually make the SLC layout rather pleasant, or what they did with Python in the Efteling. There's exceptions to the rules shamefully enough

**Reply by TheAtomTwister** [reply]
It is true, there are stronger versions of the Arrow track now.  Arrow itself made a new version of its track for Tennessee Tornado, one of its last rides.  There are subtle differences in track appearance with those though.  Perhaps there can be multiple versions of the same ride track that get researched at different times, smoother counterparts of different track.

**Reply by Sadret** [reply]
@TheAtomTwister The changes you suggest are probably too complex to implement. How would you determine how rough a ride is?
Also, how much does it really matter for the gameplay?

**Reply by TheAtomTwister** [reply]
Oh no, the knowledge of how rough a ride is in real life gives you an idea of what to set the coefficients to for the factors that contribute to intensity in the game.  Some of the coefficients can rise based on the rides age.  That can be coded in pretty easily.

**Reply by TheAtomTwister** [reply]
The idea is not too difficult to do if you're rebuilding the excitement-intensity-nausea (EIN) algorithms from the ground up, which is what X is suggesting here.  I actually think that X's idea of rebuilding them from scratch is a really good one.  If I were coding OpenRCT2 instead of Minecraft, I'd probably be tempted to do it myself.  It sounds like fun.

**Comment by ageekhere** [comment]
With Cash machines I would cap the withdraw to $20 and every time a peep uses the Cash machines they became harder and harder to please
e.g
Happiness drops quicker 
They want more exciting rides
Are more picky on the price of rides
Gets bored of one type of food/drink

Until they are too hard to please and they leave.
